Tests/Unit/EmogrifierTest.php

```diff
@@ -1090,4 +1090,46 @@ public function emogrifyMergesCssWithMixedUnits()
             $this->subject->emogrify()
         );
     }
+
+    /**
+     * @test
+     */
+    public function emogrifyRemovesInvisibleElements()
+    {
+        $css = 'div.foo { display: none; }';
+        $html = $this->html5DocumentType . self::LF .
+            '<html><body><div class="bar"></div><div class="foobar" style="display: none;"></div>'
```

Comment:
Actually, these should be two separate tests: one for the case that the display: none comes from the CSS via setCss, and one for the case that the display: none comes from the style attribute. I propose the following two test method names:

emogrifyByDefaultRemovesElementsWithDisplayNoneInStyleAttribute
emogrifyByDefaultRemovesElementsWithDisplayNoneFromExternalCss

What do you think?